packages/build-tools/src/steps/functions/sendSlackMessage.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  BuildFunction,
+  BuildStepContext,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+} from '@expo/steps';
+import { BuildStepInputById } from '@expo/steps/dist_esm/BuildStepInput';
+
+export function createSendSlackMessageFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'send_slack_message',
+    name: 'Send Slack message',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'slack_hook_url',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: true,
+      }),
+      BuildStepInput.createProvider({
+        id: 'message',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: true,
+      }),
+    ],
+    fn: async (stepCtx, { inputs }) => {
+      await sendSlackMessageAsync(stepCtx, inputs);
+    },
+  });
+}
+
+export async function sendSlackMessageAsync(
+  stepCtx: BuildStepContext,
+  inputs: BuildStepInputById
+): Promise<void> {
+  const { logger } = stepCtx;
+  const slackHookUrl = inputs.slack_hook_url.value as string;
+  const slackMessage = inputs.message.value as string;
+  logger.info('Sending Slack message');
+
+  const body = { text: slackMessage };
+  let fetchResult: Response;
+  try {
+    fetchResult = await fetch(slackHookUrl, {
+      method: 'POST',
+      body: JSON.stringify(body),
+      headers: { 'Content-Type': 'application/json' },
+    });
+  } catch (error) {
+    logger.warn(`Sending Slack message to webhook url "${slackHookUrl}" failed`);
+    logger.debug(error);
+    throw new Error(`Sending Slack message to webhook url "${slackHookUrl}" failed`);
```

Comment:
Can users see this debug log? If not maybe we should log the underlying error better 🤔? 

Can this error contain secret information we don't want to put in the logs? I don't think so. If somebody considers the webhook URL a secret it will be masked anyway.

Reply:
Haven't tested it yet, I'll check

Reply:
User sees the `warn` and the `error`, not the `debug`. I thought of it more as a way for us to find the true error in the logs if ever there will be user asking why his slack messaging doesn't work. But if you think exposing the contents of the error in their entirety is safe, I can do that too